NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,235 @@
+        addSubview(scrollView)
+    }
+    
+    public init(items: [String]) {
+        self.buttonTitles = items
+        
+        super.init(frame: CGRect.zero)
+        
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+        createButtons()
+    }
+    
+    public func setItems(_ items: [String]) {
+        self.buttonTitles = items
+        createButtons()
+        setNeedsLayout()
+    }
+    
+    // Layout subviews - set frames
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        scrollView.frame = CGRect(x: 0, y: 0, width: frame.size.width, height: frame.size.height)
+        
+        updateUI()
+        drawSeparatorLine()
+        highlightSelectedSegment()
+        calculateAndSetScrollOffset()
+    }
+    
+    // Action methods
+    @objc private func buttonTapped(_ sender: UIButton) {
+        let previousButton = buttonsArray.filter{$0.tag == selectedSegmentIndex}.first
+        
+        selectedSegmentIndex = sender.tag
+        
+        if previousButton?.tag != sender.tag {
+            sendActions(for: .valueChanged)
+            previousButton?.isSelected = false
+            UIButton.animate(withDuration: 0.3, animations: {
+                previousButton?.transform = CGAffineTransform.identity
+            })
+        }
+ 
+        sender.isSelected = true
+    }
+    
+// Private
+    // Create buttons
+    private func createButtons() {
+        for button in buttonsArray {
+            button.removeFromSuperview()
+        }
+        buttonsArray.removeAll()
+        
+        var buttonTag: Int = 0
+        
+        for title in buttonTitles {
+            let button = UIButton(type: .custom)
+        
+            button.setTitleColor(UIColor.lightGray, for: .normal)
+            button.setTitleColor(UIColor.black, for: .selected)
+            button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+            button.setTitle(title, for: .normal)
+            
+            button.tag = buttonTag
+            
+            buttonTag += 1
+            
+            buttonsArray.append(button)
+        }
+    }
+    
+    private func updateUI() {
+        var numberOfPaddings: CGFloat = -1
+        var buttonX: CGFloat = buttonMargin
+        var contentSizeWidth: CGFloat = 0
+        
+        // Set button frame
+        for button in buttonsArray {
+            button.transform = CGAffineTransform.identity
+            button.titleLabel?.font = font
+            
+            let buttonSize = calculateButtonSize(title: (button.titleLabel?.text)!, fontAttributes: [NSAttributedStringKey.font: font!])
+            
+            button.frame = CGRect(x: buttonX,
+                                  y: (frame.size.height - buttonSize.height) / 2,
+                                  width: buttonSize.width,
+                                  height: buttonSize.height)
```

Comment:
Button text is vertically centred by default. You could simplify by setting y to 0, and height to scroll view height.